src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1109,29 +1059,33 @@ void Ekf2::run()
 					}
 				}
 
-				// Check and save the last valid calibration when we are disarmed
-				if ((vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_STANDBY)
-				    && (status.filter_fault_flags == 0)
+				// Check and update the last valid magnetometer calibration
```

Comment:
Note that the parameter save is still rate limited and only happens on land and disarm.